otherlibs/dune-rpc/private/where.ml

```diff
@@ -151,20 +151,22 @@ module Make
          | Ok s -> Ok (Some s)
          | Error exn -> Error exn)
     | None ->
-      let of_file f =
-        let+ contents = IO.read_file f in
-        match contents with
-        | Error e -> Error e
-        | Ok contents ->
-          (match of_string contents with
+      (match default ~build_dir () with
```

Comment:
I don't know what the issue is yet, but this fix doesn't seem right to me. Previously, the code would look at the socket inside the build directory to connect, and now it's just assuming the address is the default address.

Reply:
Nothing has changed here for Unix systems: `default ~build_dir ()` still returns the socket inside the build directory:

https://github.com/ocaml/dune/blob/071cb4718d4c23f9b3645c8c254028df6d1a35c2/otherlibs/dune-rpc/private/where.ml#L130-L134

Reply:
Okay, but even on Windows, the workflow to connect should be:

1. Inspect the file _build/.rpc/dune
2. If it's a socket, connect to it (will never happen on windows)
3. If it's a normal file, read it and get the connection string from here. On windows, that connection string will have the address we want to connect on.

Reply:
Oops, you are right. I hadn't grasped the logic behind the client/server split. Now I pushed a commit with the right fix I think.